Add function to compute value at a given return period

mbruggs opened the pull request:
This PR adds a function called `return_year_value`. An example usage can be seen in the `extreme_response_full_sea_state_example.ipynb`. The example would be adjusted from:

![image](https://user-images.githubusercontent.com/13438942/192002316-90682835-46e9-4ce1-85b5-05af149545a5.png)


To:
![image](https://user-images.githubusercontent.com/13438942/192002394-f4572488-7a60-4aaf-a771-19d57433c3a8.png)


With the `return_year_value` function defined as follows:
```
def return_year_value(ppf, return_year, short_term_period_hr):
    """
        Calculate the value from a given distribution corresponding to
        a particular return year.
        Parameters
        ----------
        ppf: callable function of 1 argument
            Percentage Point Function (inverse CDF) of short term distribution.
        return_year: int
            Return period in years.
        short_term_period_hr: int, float
            Short term period the distribution is created from in hours.
        Returns
        -------
        value: float
            The value corresponding to the return period from the distribution.
    """
    assert callable(ppf)
    assert isinstance(return_year, int)
    assert isinstance(short_term_period_hr, (float, int))

    p = 1 / (return_year * 365.25 * 24 / short_term_period_hr)

    return ppf(1 - p)
```

mbruggs commented:
Thanks for looking @ssolson! Sorry for not adding the test, I completely missed the test directory when I looked. I'll add a test once we have some feedback from Carlos.

I've run autopep8 over the function and pushed the changes but let me know what other formatting changes are needed.

cmichelenstrofer commented:
@ssolson Yes this is very useful. I was doing it manually in the examples but having a function is better.

cmichelenstrofer commented on examples/extreme_response_full_sea_state_example.ipynb:
```diff
@@ -493,7 +493,7 @@
    "metadata": {},
    "source": [
-    "Finally we plot the survival function and show the 100-year return level (dashed grey lines). The 100-year value is about 104 m (where the grey line intersects the x-axis)"
+    "Finally we plot the survival function and show the 100-year return level (dashed grey lines). The 100-year value is about 11.1 m (where the grey line intersects the x-axis)"
```
104m lol :) Glad someone else looked at this example!

cmichelenstrofer commented on mhkit/loads/extreme.py:
```diff
@@ -727,3 +727,31 @@ def mler_export_time_series(rao, mler, sim, k):
+
+    p = 1 / (return_year * 365.25 * 24 / short_term_period_hr)
+
+    return ppf(1 - p)
```
Great!

cmichelenstrofer reviewed (approved):
Looks great, thanks for doing this!

akeeste commented:
@mbruggs Carlos approved your changes. Can you add a test for this function? There are plenty of examples in [/tests/test_loads/](https://github.com/MHKiT-Software/MHKiT-Python/blob/master/mhkit/tests/test_loads.py). I'm also going to change this PR to go into our Develop branch, per our current workflow. @ssolson or I can do a final review once the test is in. 

Adam

mbruggs commented:
Thanks for the input everyone, much appreciated. I've added a test but it found it hard to know exactly what the test should be. In the end, the test is against a 'known value', ie computed with the function under test itself. This should, at least, guard against any unexpected changes in the function signature or calculation, assuming that the current calculation is correct.

Sorry for missing that the branch should be merged into Develop. I've rebased my branch against Develop so the merge should be neat.

Thanks again for all the work on MHKiT!

akeeste commented on mhkit/tests/loads/test_loads.py:
```diff
@@ -180,6 +180,20 @@ def test_mler_export_time_series(self):
+
+        val = loads.extreme.return_year_value(dist.ppf, return_year, short_term_period)
+
+        # This is a little hard to test as there is not an indepedent means by
+        # which we can do this calculation. The value below has been taken from
+        # an initial calculation and will indicate if the return_year_value is
+        # changed unexpectedly.
```
I think this is acceptable right now, I can think of a couple other tests that do something similar. @ssolson or @cmichelenstrofer thoughts on alternative known values?

Can you remove the comment lines, those will be recorded in this thread.

cmichelenstrofer replied:
This is a good enough test.

mbruggs replied:
Thanks both for confirming.

akeeste commented on mhkit/loads/extreme.py:
```diff
@@ -727,3 +727,31 @@ def mler_export_time_series(rao, mler, sim, k):
+    ----------
+    ppf: callable function of 1 argument
+        Percentage Point Function (inverse CDF) of short term distribution.
+    return_year: int
```
Does ``return_year`` have to be an int? I think we could allow floats here as well without issue.

mbruggs replied:
Good point, I agree that it could be a float. Updated.

akeeste reviewed (changes requested):
@mbruggs I made a few minor clean-up comments. Otherwise the function is appropriately short and clean. I don't have much else, aside from getting a second opinion from @cmichelenstrofer or @ssolson on what the test is against.
